sql/hive-thriftserver/src/main/java/org/apache/hive/service/cli/thrift/ThriftHttpServlet.java

```diff
@@ -546,14 +546,15 @@ private boolean isKerberosAuthMode(String authType) {
     return authType.equalsIgnoreCase(HiveAuthFactory.AuthTypes.KERBEROS.toString());
   }
 
-  private static String getDoAsQueryParam(String queryString) {
+  private static String getDoAsQueryParam(HttpServletRequest request) {
+    String queryString = request.getQueryString();
     if (LOG.isDebugEnabled()) {
       LOG.debug("URL query string:" + queryString);
     }
     if (queryString == null) {
       return null;
     }
-    Map<String, String[]> params = jakarta.servlet.http.HttpUtils.parseQueryString( queryString );
```

Comment:
`HttpUtils` was deprecated in Servlet 5.0 and removed in 6.0.
https://jakarta.ee/specifications/servlet/5.0/apidocs/jakarta/servlet/http/httputils

Reply:
OK, I will open another one.